### ISSUE TITLE
Modal dialogs are not accessible

### ISSUE THREAD

**Issue by citelao** [dart-lang/angular_components#239]
I'm working to improve accessibility in an unrelated project, and [I investigated](https://github.com/filiph/edgehead/issues/12#issuecomment-375796283) Angular components' modal dialog.

The [modal dialogs](https://github.com/dart-lang/angular_components/blob/master/lib/material_dialog) are not accessible at the moment, but they should be straightforward to fix.

Specifically, your modal should:
- announce that it is a dialog
- trap screen reader focus with in it
- be dismissable with `ESC` and announce this to screen reader users.

This should be attainable by:
1. marking your dialog as `role="dialog"`
4. adding `labelledby` and `describedby` as appropriate to the dialog (dismissal via esc should be indicated here)
2. adding a `role="document"` wrapper to the dialog content (note: this needs to be investigated further)
3. adding `ESC` keybindings to dismiss the modal

I recorded a [video demonstration](https://www.youtube.com/watch?v=SMqC_Z6DSak&feature=youtu.be) comparing the [Angular modal](https://dart-lang.github.io/angular_components_example/#/material_dialog) to an (admittedly buggy) [implementation I like](https://gdkraus.github.io/accessible-modal-dialog/). Note how on activation, screen reader users immediately know they are in a dialog and get context, even though the focus is somewhere within the dialog. Also note how the screen reader cannot leave the dialog.

Let me know how I can help!

**Comment by TedSander** [comment]
Thanks @citelao we are doing a lot of accessibility work right now and this will be helpful.

**Comment by citelao** [comment]
Glad I could help!

**Comment by filiph** [comment]
Hi,

some of this work has already landed in (or before) the current version, 0.9.0. Not sure if it covers (all) the issues you raised.

cc @nshahan

**Comment by citelao** [comment]
Are the demos live?

**Comment by nshahan** [comment]
Yes, the live examples have been updated. https://dart-lang.github.io/angular_components_example/

Although, I don't think the specific fixes you mentioned for the Material Dialog component have been implemented yet.

**Comment by TedSander** [comment]
We improved the a11y here. Should be working now.
